Tasks/DownloadBuildArtifacts/task.json

```diff
@@ -30,30 +30,55 @@
       "type": "pickList",
       "label": "Build definition",
       "defaultValue": "",
-      "required": true,
+      "required": false,
       "properties": {
         "EditableOptions": "True"
       },
       "helpMarkDown": "Select the build definition name"
     },
     {
-      "name": "version",
+      "name": "buildId",
       "type": "pickList",
       "label": "Build",
-      "defaultValue": "",
+      "defaultValue": "$(Build.BuildId)",
       "required": true,
       "properties": {
         "EditableOptions": "True"
       },
       "helpMarkDown": "Select the build number"
     },
```

Comment:
The experience in both Build and RM can be better if there was first an option to choose between "The current build" and "Specified build of a definition". If the current build is chosen, it implies Build.BuildId, and one does not have to worry about "Project", "Build definition", and "Build". If "Specified build of a definition" is chosen, then user would have to pick these three values. In the manner it is coded, the most common scenario is made more complex by exposing many properties both in RM and Build than what they are used to currently.

Note that we would expand "Specified build of a definition" in the future. Instead of selecting a specific build number, can use say "latest from a branch", etc. So, the above change would also be aligned with that.

Reply:
Can release have artifacts now?  I wonder if there is an option where the Download from current can work for both build and release.

Reply:
It is not really "current" in RM, it is more of "primary" build. In other words, the intent is to download from the same build in Build, and from the primary referenced build in RM. Not sure what the text of the field should be.

Reply:
done, added and option to choose between current build and specific build.